addons/Magic_Boxes/src/presenter.js

```diff
@@ -1264,6 +1326,91 @@ function AddonMagic_Boxes_create() {
         presenter.isSelectionPossible = true;
 
         presenter.isShowAnswersActive = false;
+        presenter.isGradualShowAnswersActive = false;
+    };
+
+    presenter.gradualShowAnswers = function () {
+        if(!presenter.isGradualShowAnswersActive){
+            presenter.saveUserSelectionBeforeShowAnswers();
+        }
+
+        presenter.isShowAnswersActive = true;
+        presenter.isGradualShowAnswersActive = true;
+        presenter.isSelectionPossible = false;
+        
+
+        if (presenter.configuration.checkByWords){
```

Comment:
Początki funkcji showNextWord i showNextLetter są identyczne, aby uniknąć duplikacji kodu fajnie by było gdybyś zmodyfikował ten fragment, aby zawierał ten duplikowany kod np:
```suggestion
        for (let key in presenter.GSAcorrectAnswerLocations){
            if (usedGSAKeys.includes(key)) continue;
            else {
                presenter.configuration.checkByWords
                    ? presenter.showNextWord(key)
                    : presenter.showNextLetter(key);
            }
```